### PR TITLE
Added examples for require and modules part

### PR DESCRIPTION
**Pull request by AdriVanHoudt**
* This definitely needs a review. All feedback welcome!

* I added this because I found myself super confused about what exactly goes where so I hope this will help, mostly new, people.

### CODE REVIEW

**Comment by yonjah** [comment]
I'm not really sure if I understand the guide correctly, but I think the right and wrong sections should be made more clear.
The code as is wont work, and the `automatically` function is part of the `internals` declaration so I guess the code should be similar to this -
```javascript
// Right
var internals = {
    package: require('./package.json'),
    foo: 'bar',
    automatically: function () {}
};

internals.automatically();

// Also right
var internals = {};

inernals.package = require('./package.json');
internals.foo = 'bar';
internals.automatically = function () {};

internals.automatically();
```

Notice I removed Hapi and Hoek requirements since they are irrelevant to the demo.

The wrong example it should be more clear what actually wrong here.
Is it using `Foo` param with capitals ? 
Is it missing the `package` param in internals ? 
Is it not actually adding the real `foo` variable as a param and just redefining the `bar` string ?

You should either add a comment to the demo or better yet just make one clear wrong example with the most common case where people might be confused about ( I would think omitting a const or global var  is will be more confusing than not attaching the package json file)

So maybe something like this -
```javascript
// Right
var internals = {
    package: require('./package.json'),
    foo: 'bar',
    automatically: function () { console.log(internals.foo); }
};

internals.automatically();

// Also right
var internals = {};

inernals.package = require('./package.json');
internals.foo = 'bar';
internals.automatically = function () { console.log(internals.foo); };

internals.automatically();

// Wrong
var foo = 'bar';
var internals = {
    package: require('./package.json'),
    automatically: function () { console.log(foo); }
};

internals.automatically();
```

**Comment by AdriVanHoudt** [comment]
Thanks for the feedback!
And this PR is to make the guide understandably ;)

I agree that the `automatically` function should be declared before the execution but that goes against the guide...

I'm not a fan of declaring the function inside the internals object thought e.g. 
```javascript
var internals = {
    auto: function() {}
};
```` 

The Hapi and Hoek requires are a demo of `### Require`.
The point was to make it clear that modules can be outside of internals. Otherwise you would go `internals.hapi = require('hapi')`.

You're right about the `wrong` example, it should be more clear. I'll update and let me know if it is better.

**Comment by AdriVanHoudt** [comment]
@yonjah @hueniverse recently made a clear example on how it should be done here https://github.com/hueniverse/hueniversity/blob/master/lib/index.js

File: Style.md [hunk 1]
```diff
@@ -671,11 +671,51 @@
...
+  var Package = require('./package.json');
+  
+  var internals = {
+      package: Package,
+      foo: 'bar'
+  };
+  
+  internals.automaticly();
+  
+  internals.automaticly = function () {};
```
Comment (hueniverse) [inline Style.md]: This should be before invoking the method.

**Pull request merged**